File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/commands/MakeSeatReservation.cs
using System;
using System.Collections.Generic;
using ENode.Commanding;

namespace ConferenceManagement.Commands
{
    public class MakeSeatReservation : Command<Guid>
    {
        public Guid ReservationId { get; set; }
        public IEnumerable<SeatReservationItemInfo> Seats { get; set; }

        public MakeSeatReservation() { }
        public MakeSeatReservation(Guid conferenceId) : base(conferenceId)
        {
            this.Seats = new List<SeatReservationItemInfo>();
        }
    }
}

File: samples/conference/common/src/main/java/com/microsoft/conference/common/management/commands/MakeSeatReservation.java
package com.microsoft.conference.common.management.commands;

import org.enodeframework.commanding.Command;

import java.util.ArrayList;
import java.util.List;

public class MakeSeatReservation extends Command<String> {
    public String reservationId;
    public List<SeatReservationItemInfo> seats;

    public MakeSeatReservation() {
    }

    public MakeSeatReservation(String conferenceId) {
        super(conferenceId);
        this.seats = new ArrayList<>();
    }
}
